deployment/README.md

```diff
@@ -12,4 +12,5 @@ The manifests in this are generated from the Helm chart automatically.
 The `values.yaml` files used to configure `keel` can be found in
 [`values`](./values/).
 
+<!-- Deprecated -->
```

Comment:
I was considering removing the deployment directory entirely because it doesn't seem to be maintained (at least no longer having the gen-deploy.sh script, nor the values.yaml). But I ultimately decided against it, since that'd be outside the scope of this PR.